protocol/streaming/full_node_streaming_manager.go

```diff
@@ -187,11 +220,68 @@ func (sm *FullNodeStreamingManagerImpl) getNextAvailableSubscriptionId() uint32
 	return id
 }
 
+// Filter StreamUpdates for subaccountIdNumbers
+// If a StreamUpdate_OrderUpdate contains no updates for subscribed subaccounts, drop message
+// If a StreamUpdate_OrderUpdate contains updates for subscribed subaccounts, construct a new
+// StreamUpdate_OrderUpdate with updates only for subscribed subaccounts
+func (sub *OrderbookSubscription) FilterSubaccountStreamUpdates(
+	output chan []clobtypes.StreamUpdate,
+	logger log.Logger,
+) {
+	subaccountIdNumbers := make([]uint32, len(sub.subaccountIds))
+	for i, subaccountId := range sub.subaccountIds {
+		subaccountIdNumbers[i] = subaccountId.Number
+	}
+
+	// If reflection becomes too expensive, split updatesChannel by message type
+	for updates := range sub.updatesChannel {
+		filteredUpdates := []clobtypes.StreamUpdate{}
+		for _, update := range updates {
+			switch updateMessage := update.UpdateMessage.(type) {
```

Comment:
For modularity, move switch statements into a new function:

`func FilterUpdates(input []clobtypes.StreamUpdate) (filteredUpdates []clobtypes.StreamUpdate)`